### PR TITLE
chore: add python to dependabot tracker

### PR DESCRIPTION
**Pull request by Bento007**
## Reason for Change

- So keeping are dependencies up-to-date is easier.

## Changes

- add dependabot configuration to track python requirements.
- move dependencies to their own directory to be compatible with dependabot
- update dockerfiles to use new requirements path.

## Notes for Reviewer
- depends on #6424

### CODE REVIEW

File: .dockerignore [hunk 1]
```diff
@@ -3,13 +3,13 @@
 
 # Allow files and directories
 !/requirements.txt
-!/requirements-backend.txt
-!/requirements-processing.txt
-!/requirements-submissions.txt
+!/python_dependecies/backend/requirements.txt
```
Comment (ebezzi) [inline .dockerignore:6]: Typo: `python_dependecies` -> `python_dependencies`

Comment (ebezzi) [reply .dockerignore:6]: Also, any reason for the underscore? Normally dashes are standard.

**Review by ebezzi — CHANGES_REQUESTED** [review]

File: .dockerignore (hunk 1 above)
Comment (Bento007) [reply .dockerignore]: I see a mix of `_` and `-` used throughout our naming schema for files and directories. `_` are python friendly so it's my default.

**Comment by prathapsridharan** [comment]
@Bento007 - Is this [PR](https://github.com/chanzuckerberg/single-cell-data-portal/pull/6424) redundant since in the `dependabot.yml` in this PR you already have checking for `npm` packages for `frontend`?

**Comment by prathapsridharan** [comment]
@Bento007 - Also because these are going to open PRs when new package versions get released, would it be worthwhile to alert the slack channel now about this?

**Comment by Bento007** [comment]
> @Bento007 - Is this [PR](https://github.com/chanzuckerberg/single-cell-data-portal/pull/6424) redundant since in the `dependabot.yml` in this PR you already have checking for `npm` packages for `frontend`

That would be a future PR

**Comment by Bento007** [comment]
> @Bento007 - Also because these are going to open PRs when new package versions get released, would it be worthwhile to alert the slack channel now about this?

that could be future work https://github.com/chanzuckerberg/single-cell-data-portal/issues/6430

**Review by ebezzi — APPROVED** [review]

**Pull request merged**